apps/lib/apps_ui.c

```diff
@@ -54,6 +54,8 @@ static int ui_read(UI *ui, UI_STRING *uis)
     reader = UI_method_get_reader(ui_fallback_method);
     if (reader)
         return reader(ui, uis);
+    /* Default to the empty password if we've got nothing better */
+    UI_set_result(ui, uis, "");
```

Comment:
I'm not an expert on the password UI but according to the explanation you gave this looks good.